string control_state
string operation_mode